### PR TITLE
enum_to_str: convert enum keys, not just enum values

### LINKED ISSUE #531: OmegaConf.to_yaml does not convert Enum DictConfig keys to string
```python3
from enum import Enum
from omegaconf import OmegaConf

class Color(Enum):
    RED = 1
    BLUE = 2

conf = OmegaConf.create({Color.RED: Color.BLUE})

print(OmegaConf.to_yaml(conf))
```
Outputs:
```
? !!python/object/apply:__main__.Color
- 1
: BLUE
```
Expected output:
```
RED: BLUE
```

Note that the DictConfig value `Color.BLUE` *is* converted to a string, but the key `Color.RED` is not.

### LINKED ISSUE #550: Behavior change: to_container(..., enum_to_str=True) to convert both keys and values
```python
$ cat tmp.py
from enum import Enum
from omegaconf import OmegaConf


class Color(Enum):
    RED = 1
    BLUE = 2


conf = OmegaConf.create({Color.RED: Color.BLUE})
print(OmegaConf.to_container(conf, enum_to_str=True))

$ python tmp.py
{<Color.RED: 1>: 'BLUE'}
```

Desired output is for both key and value to be converted when `enum_to_str` is set to True:
```
{'RED': 'BLUE'}
```

### PR DESCRIPTION
**Pull request by Jasha10**
Closes #531
Closes #550

### CODE REVIEW

**Comment by Jasha10** [comment]
I created some tests for the `enum_to_str` argument to `to_container`.
Previously there were no tests for this.

File: tests/test_to_container.py [hunk 1]
```diff
@@ -167,3 +167,121 @@ def test_string_interpolation_with_readonly_parent() -> None:
...
+                id="int_key:F-unaffected",
+            ),
+        ],
+    )
+    def test_enum_to_str_for_keys(
+        self, src: DictConfig, enum_to_str: bool, expected: Any
+    ) -> None:
+        container = OmegaConf.to_container(src, enum_to_str=enum_to_str)
+        assert isinstance(container, dict)
+        key = list(container.keys())[0]
```
Comment (omry) [inline tests/test_to_container.py:209]: If you use the same key name in all inputs you would be able to use it directly here.

File: tests/test_to_container.py [hunk 2]
```diff
@@ -167,3 +167,121 @@ def test_string_interpolation_with_readonly_parent() -> None:
 def test_to_container_missing_inter_no_resolve(src: Any, expected: Any) -> None:
     res = OmegaConf.to_container(src, resolve=False)
     assert res == expected
+
```
Comment (omry) [inline tests/test_to_container.py:170]: That's a bit too many tests.
Let's try to keep tests high value, a lot of test code becomes a maintenance burden.

I think we can go with a single test function testing:
1. with and without enum_to_str
2. A single config containing dict and list, enums as keys and enums as values.

Maybe:
```python
cfg = {
  "d": {
    Color.RED: "red",
    "red" : Color.RED,   
  },
  "l" : ["red", Color.RED]
}
```

Testing conversion of a config derived from this, with the flag true and false should give solid coverage and will be much more compact.

Comment (Jasha10) [reply tests/test_to_container.py:170]: Done in https://github.com/omry/omegaconf/pull/549/commits/812ce8018e466308ef80d6ad3ab7812971c86da6

File: tests/test_to_container.py (hunk 1 above)
Comment (Jasha10) [reply tests/test_to_container.py:187]: Done in https://github.com/omry/omegaconf/pull/549/commits/910e4d9df9acfee9738ab3226199b855936a13d0.

File: tests/test_to_container.py [hunk 3]
```diff
@@ -167,3 +167,55 @@ def test_string_interpolation_with_readonly_parent() -> None:
...
+        ],
+    )
+    def test_enum_to_str_for_list(
+        self, src: Any, enum_to_str: bool, expected: Any
+    ) -> None:
+        cfg = OmegaConf.create(src)
+        container: List[Any] = OmegaConf.to_container(cfg, enum_to_str=enum_to_str)  # type: ignore
+        value = container[0]
+        assert value == expected
+        assert type(value) == type(expected)
```
Comment (omry) [inline tests/test_to_container.py:221]: ditto.

**Review by omry — COMMENTED** [review]
looking good. a few nits.

**Comment by omry** [comment]
Oh, I thought about #550 a bit:
Every bugfix is a behavior change, I don't think this one justifies an API change entry. (I doubt anyone have actually ran into it).

**Comment by Jasha10** [comment]
> I don't think this one justifies an API change entry.

Ok, I'll scrap the #550 news fragment.

File: tests/test_to_container.py (hunk 3 above)
Comment (Jasha10) [inline tests/test_to_container.py:221]:
```suggestion
        assert container == [expected]
```

**Review by omry — APPROVED** [review]
Great, thanks.

**Pull request merged**